contracts/campaign.sol

```diff
@@ -460,41 +501,53 @@ contract campaign is oracleClient {
         o.askBounty(typeSN, idPost, idUser, idProm);
     }
 
+/**
+ * @dev Updates the bounty payment status for a specific campaign promotion (idProm).
+ * The function calculates the gain for the influencer based on the number of subscribers (nbAbos) and campaign bounties.
+ * If the gain is sufficient, the payment is processed, otherwise, the funds are added to the promotion's balance.
+ * @param idProm The unique identifier of the campaign promotion.
+ * @param nbAbos The number of subscribers for the promotion.
+ * @return ok True if the operation is successful.
+ */
     function updateBounty(bytes32 idProm, uint256 nbAbos)

```

Comment:
Combiner les deux fonctions "updateBounty" et "update".

function updateBountyAndPost(
    bytes32 idProm,
    uint256 nbAbos,
    bytes32 idRequest,
    uint64 likes,
    uint64 shares,
    uint64 views
) external notPaused returns (bool ok) {
    require(msg.sender == oracle, "oracle mismatch");

    // Store the prom element for efficient access
    SocialMediaPostData storage post = proms[idProm];
    require(!post.isPayed, "link already paid");
    post.isPayed = true;
    
    // Store the campaign element for efficient access
    AdPool storage campaignData = campaigns[post.idCampaign];
    post.funds.token = campaignData.funds.token;

    uint256 bountyGain = 0;
    for (uint256 i = 0; i < campaignData.bounties.length; i++) {
        if (
            nbAbos >= campaignData.bounties[i].minRange &&
            nbAbos < campaignData.bounties[i].maxRange &&
            post.typeSN == campaignData.bounties[i].typeSN
        ) {
            bountyGain = campaignData.bounties[i].amount;
        }
    }

    uint256 postGain = 0;

    if (likes > results[post.prevResult].likes) {
        postGain +=
            (likes - results[post.prevResult].likes) *
            campaigns[post.idCampaign].ratios[post.typeSN].likeRatio;
    }
    if (shares > results[post.prevResult].shares) {
        postGain +=
            (shares - results[post.prevResult].shares) *
            campaigns[post.idCampaign].ratios[post.typeSN].shareRatio;
    }
    if (views > results[post.prevResult].views) {
        postGain +=
            (views - results[post.prevResult].views) *
            campaigns[post.idCampaign].ratios[post.typeSN].viewRatio;
    }
    post.prevResult = idRequest;

    if (campaignData.funds.amount <= bountyGain) {
        post.funds.amount += campaignData.funds.amount;
        campaignData.funds.amount = 0;
        emit CampaignFundsSpent(post.idCampaign);
    } else {
        campaignData.funds.amount -= bountyGain;
        post.funds.amount += bountyGain;
    }

    if (post.funds.token == address(0)) {
        post.funds.token = campaigns[post.idCampaign].funds.token;
    }
    if (campaigns[post.idCampaign].funds.amount <= postGain) {
        post.funds.amount += campaigns[post.idCampaign].funds.amount;
        campaigns[post.idCampaign].funds.amount = 0;
        emit CampaignFundsSpent(post.idCampaign);
    } else {
        campaigns[post.idCampaign].funds.amount -= postGain;
        post.funds.amount += postGain;
    }

    return true;
}

Reply:
combining the two functions like you did won't work in this way 'cause on is used for bouty campaign & the other for performance